### PR TITLE
Add StudentId attribute to model

### PR DESCRIPTION
**Pull request by channne**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/StudentId.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class StudentId {
+
+    public static final String MESSAGE_CONSTRAINTS = "Student Numbers should start with an A, "
+            + "followed by 7 numbers, then end with an alphabet.";
+
+    /*
+     * The first character of the Student ID must be an A,
+     * followed by 7 digits, then an alphabet.
```
Comment (bryans17) [inline src/main/java/seedu/address/model/person/StudentId.java:13]: Maybe we should emphasize that it must end with a capital alphabet.

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 2]
```diff
@@ -107,87 +115,116 @@ public void parse_compulsoryFieldMissing_failure() {
...
 
         // two invalid values, only first invalid value reported
         assertParseFailure(parser, INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB + INVALID_ADDRESS_DESC
-                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB, Name.MESSAGE_CONSTRAINTS);
+                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + STUDENTID_DESC_BOB, Name.MESSAGE_CONSTRAINTS);
 
         // non-empty preamble
         assertParseFailure(parser, PREAMBLE_NON_EMPTY + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
+                + ADDRESS_DESC_BOB + GITHUB_DESC_BOB + STUDENTID_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
```
Comment (bryans17) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:227]: Accidentally overwrote the TELEGRAM_DESC_BOB string.

**Review by bryans17 — COMMENTED** [review]
Just need to update 2 minor issues 😄

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 2 above)
Comment (channne) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:227]: Nice catch!

File: src/main/java/seedu/address/model/person/StudentId.java (hunk 1 above)
Comment (channne) [reply src/main/java/seedu/address/model/person/StudentId.java:13]: Okay!

**Review by bryans17 — APPROVED** [review]
LGTM 👍🏻.

Closes #43.

**Pull request merged**